custom/icds_reports/utils/__init__.py

```diff
@@ -1774,3 +1777,81 @@ def create_group_by(aggregation_level):
 
 def column_value_as_per_agg_level(aggregation_level, agg_level_threshold, true_value, false_value):
     return true_value if aggregation_level > agg_level_threshold else false_value
+
+
+@attr.s
+class AggLevelInfo(object):
+    agg_level = attr.ib()
+    col_name = attr.ib()
+
+
+def _construct_repacement_map_from_awc_location(loc_level, replacement_location_ids):
+    levels = {
+        'state': AggLevelInfo(1, 'state'),
+        'district': AggLevelInfo(2, 'district'),
+        'block': AggLevelInfo(3, 'block'),
+        'supervisor': AggLevelInfo(4, 'supervisor'),
+        'awc': AggLevelInfo(5, 'doc')
+    }
+    level_info = levels[loc_level]
+    filters = {
+        f'{level_info.col_name}_id__in': replacement_location_ids,
+        'aggregation_level': level_info.agg_level
+    }
+    columns = [
+        f'{level_info.col_name}_id',
+        'state_name',
+        'district_name',
+        'block_name',
+        'supervisor_name',
+        'awc_name'
+    ]
+    replacements = AwcLocation.objects.filter(**filters).values(*columns)
+
+    def _full_hierarchy_name(loc):
+        loc_names = [loc[f'{level}_name'] for level in levels.keys() if loc[f'{level}_name']]
+        return ' > '.join(loc_names)
+
+    replacement_names = {loc[f'{level_info.col_name}_id']: _full_hierarchy_name(loc) for loc in replacements}
+    return replacement_names
+
+
+def _construct_repacement_map_from_sql_location(replacement_location_ids):
+    # prefetch all possible parents
+    replacements = SQLLocation.objects.filter(location_id__in=replacement_location_ids).select_related('parent__parent__parent__parent')
+
+    def _full_hierarchy_name(loc):
+        loc_names = []
+        while loc is not None:
+            loc_names.append(loc.name)
+            loc = loc.parent
+        loc_names.reverse()
+        return ' > '.join(loc_names)
+
+    replacement_names = {loc.location_id: _full_hierarchy_name(loc) for loc in replacements}
+    return replacement_names
+
+
+def get_deprecation_info(locations, show_test, multiple_levels=False):
+    locations_list = []
+    replacement_location_ids = []
+    for loc in locations:
+        loc_level = loc.location_type.name
+        if show_test or loc.metadata.get('is_test_location', 'real') != 'test':
+            locations_list.append(loc)
+            if loc.metadata.get('deprecated_to'):
+                replacement_location_ids.extend(loc.metadata.get('deprecated_to', []))
+            if loc.metadata.get('deprecates'):
+                replacement_location_ids.extend(loc.metadata.get('deprecates', []))
+
+    if multiple_levels:
+        replacement_names = _construct_repacement_map_from_sql_location(replacement_location_ids)
+
+    else:
+        replacement_names = _construct_repacement_map_from_awc_location(loc_level, replacement_location_ids)
+
+    return locations_list, replacement_names
+
+
+def get_replacement_name(location, field, replacement_names):
```

Comment:
how about `get_location_names_from_metafield`?

Reply:
I agree this name can be improved, but the replacement names arent actually coming from the meta, they are coming from the replacment_names dict I pass in, so I don't love that option. I will think a bit on this

Reply:
Just considering this method as a standalone where it does not know the intention of the names so if we rename method name and argument as
```
def get_location_names_from_metafield(location, field, location_names)
```
it will sound better.

Reply:
How about `get_location_replacement_name` that makes it more clear that we are talking about locations, while not implying the name is in the meta field?

Reply:
That earlier name suggestion came from combining the two things
1. avoid using replacement
2. make method name descriptive of what its doing

> but the replacement names aren't actually coming from the meta

Yup, but the location id is in the metafield, so I included that in the method name.

`get_location_replacement_name` partially conveys what the method is doing which would be confusing, it is getting location name from location_names but how does it pick it? but its better than `get_replacement_name`
(this is non blocking)